From 356542324fa439de544b5e419b91ae68d42c306c Mon Sep 17 00:00:00 2001
From: Nick Wellnhofer <wellnhofer@aevum.de>
Date: Fri, 20 Jun 2025 23:46:46 +0200
Subject: [PATCH] parser: Fix xmlCtxtIsStopped

Make xmlCtxtIsStopped check for fatal errors as well. This makes it
easier to migrate away from disableSAX.
---
 parserInternals.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/parserInternals.c b/parserInternals.c
index 4e39db1f8..45085cd52 100644
--- parserInternals.c
+++ parserInternals.c
@@ -3327,7 +3327,7 @@ xmlCtxtIsStopped(xmlParserCtxtPtr ctxt) {
     if (ctxt == NULL)
         return(0);
 
-    return(PARSER_STOPPED(ctxt));
+    return(ctxt->disableSAX != 0);
 }
 
 #ifdef LIBXML_VALID_ENABLED
-- 
GitLab

